zeq/detail/broker.h

```diff
@@ -14,11 +14,12 @@ std::string buildZmqURI( std::string host, const uint16_t port )
 {
     if( host.empty( ))
         host = "*";
-    if( host != "*" && port == 0 ) // zmq does not support host:0
-        throw std::runtime_error(
-            "OS-chosen port not supported with hostname " + host );
 
-    return std::string( "tcp://" ) + host + ":" + std::to_string( int( port ));
+    const std::string tcpURI( "tcp://" + host );
+    if( host != "*" && port == 0 ) // zmq expects host:* instead of host:0
+        return tcpURI + ":*";
```

Comment:
Mmmm... should the empty host (tcp://:*) work? I don't think it does, but tcp://*:* works fine as far as I can tell